runtime/runtime.m

```diff
@@ -900,24 +1025,25 @@ -(void) xamarinSetGCHandle: (int) gc_handle;
 static void
 exception_handler (NSException *exc)
 {
+	// COOP: not accessing managed memory: any mode
 	LOG (PRODUCT ": Received unhandled ObjectiveC exception: %@ %@", [exc name], [exc reason]);
 	
-	// This might happen on a thread we haven't heard about before
-	if (mono_domain_get () == NULL)
-		mono_jit_thread_attach (NULL);
-
```

Comment:
The native-to-managed wrapper for the managed delegate (called by `xamarin_throw_ns_exception` below) should automatically attach if needed.